src/components/TextInputFocusable/index.js

```diff
@@ -121,17 +122,31 @@ class TextInputFocusable extends React.Component {
     }
 
     /**
-     * Check the paste event for an attachment and
-     * fire the callback with the selected file
+     * Check the paste event for an attachment, parse the data and
+     * call onPasteFile from props with the selected file
      *
      * @param {ClipboardEvent} event
      */
     checkForAttachment(event) {
-        if (event.clipboardData.files.length > 0) {
+        const {files, types} = event.clipboardData;
+        const TEXT_HTML = 'text/html';
+        if (files.length > 0) {
             // Prevent the default so we do not post the file name into the text box
             event.preventDefault();
             this.props.onPasteFile(event.clipboardData.files[0]);
         }
+        if (types.includes(TEXT_HTML)) {
+            const domparser = new DOMParser();
+            const embededImages = domparser.parseFromString(event.clipboardData.getData(TEXT_HTML), TEXT_HTML).images;
+            if (embededImages.length > 0) {
+                event.preventDefault();
+                fetch(embededImages[0].src)
+                    .then(x => x.blob())
+                    .then(x => new File([x], `pasted_image.${mime.extension(x.type)}`, {}))
+                    .then(this.props.onPasteFile)
+                    .catch();
```

Comment:
Rather than silently failing here, can you use the `Alert` component to show a user-friendly error message? And also let's add in a `console.debug` with any error message so we can debug any future errors more easily.

Reply:
I have implemented it via the alert not Alert.alert RN API. that is because the alert component is only for the native apps and I think this feature doesn't apply to those platforms.